tests/manual_jobs/test_job_statuses.py

```diff
@@ -0,0 +1,37 @@
+from unittest.mock import patch
+
+import pytest
+
+from snuba.manual_jobs import Job, JobLogger, JobSpec
+from snuba.manual_jobs.job_status import JobStatus
+from snuba.manual_jobs.runner import get_job_status, run_job
+
+JOB_ID = "abc1234"
+async_job_spec = JobSpec(job_id=JOB_ID, job_type="AsyncJob", is_async=True)
+
+
+class AsyncJob(Job):
+    def __init__(self, job_spec: JobSpec):
+        super().__init__(job_spec)
+
+    def execute(self, logger: JobLogger) -> None:
+        pass
+
+    @classmethod
+    def async_finished_check(cls, job_id: str) -> bool:
+        return False
+
+
+@pytest.mark.redis_db
+def test_job_status_changes_to_async_running_background() -> None:
+    assert get_job_status(JOB_ID) == JobStatus.NOT_STARTED
+    run_job(async_job_spec)
+    assert get_job_status(JOB_ID) == JobStatus.ASYNC_RUNNING_BACKGROUND
+
+
+@pytest.mark.redis_db
+def test_job_status_changes_from_async_running_background_to_finished() -> None:
+    assert get_job_status(JOB_ID) == JobStatus.NOT_STARTED
+    run_job(async_job_spec)
+    with patch.object(AsyncJob, "async_finished_check", return_value=True):
+        assert get_job_status(JOB_ID) == JobStatus.FINISHED
```

Comment:
sorry to be annoying, but could we add a third assertion here to verify this job passes through  `ASYNC_RUNNING_BACKGROUND` before reaching `FINISHED`?